### PR TITLE
select input for relationship via label_field config on destination resource POC (#209)

### PR DESCRIPTION
**Pull request by netProphET**
Addition of a `label_field` in a destination resource configuration signals to use a select box to represent the related resource.

```elixir
admin do
    label_field :admin_label
end
```

`label_field` can be a resource attribute or a calculation.

### Contributor checklist

- [ ] Bug fixes include regression tests
- [ ] Features include unit/acceptance tests

### CODE REVIEW

**Comment by netProphET** [comment]
Revised approach based on feedback to Issue #209 

If I'm on the right track, I'll continue with tests and anything else that needs tidying up or documenting. I'm still a bit new to the ecosystem, so guidance is appreciated.

**Comment by zachdaniel** [comment]
Exciting! I'll look either this weekend or on Monday 🥳

File: lib/ash_admin/components/resource/form.ex [hunk 1]
```diff
@@ -196,7 +196,11 @@ defmodule AshAdmin.Components.Resource.Form do
             class="block text-sm font-medium text-gray-700"
             for={@form.name <> "[#{attribute.name}]"}
           >
-            <%= to_name(attribute.name) %>
+            <%= if foreign_key?(@resource, attribute) do %>
+              <%= attribute.name |> to_string() |> String.replace_suffix("_id", "") |> to_name() %>
```
Comment (zachdaniel) [inline lib/ash_admin/components/resource/form.ex:200]: I would say this should turn into the relationship name that it maps to. So like `<%= if relationship = foreign_key_relationship(....)`

**Comment by zachdaniel** [comment]
So, this is very cool, but I do worry that it can only really be used for low cardinality relationships. i.e if you had 100k users, you would need a type-ahead that would do things like fetch data after a few characters were displayed etc.

If we add this and someone misuses it, it might look good in dev but then when running in prod it could cause significant issues.

**Comment by zachdaniel** [comment]
Would you be open to making it a typeahead? It doesn't have to look beautiful :)

**Comment by zachdaniel** [comment]
I think it would make sense to do one of two things:

1. fetch some limited number of results, and if there are less than that that come back, we show a select, and if there are more we show a typeahead.
2. make it configurable from the outset in the source form. Like configuring it from the source.

**Comment by netProphET** [comment]
@zachdaniel this is coming together nicely, but I'm having difficulty with, of all things, the query for typeahead suggestions

This is the current non-working state of my function

```elixir
  defp fetch_suggestions(resource, query) do
    label_field = AshAdmin.Resource.label_field(resource)
    pk_field = Ash.Resource.Info.primary_key(resource)

    resource
    |> Ash.Query.new()
    |> Ash.Query.load([pk_field, label_field])
    |> Ash.Query.filter(
      contains(
        ^label_field,
        ^%Ash.CiString{string: query}
      )
    )
    |> Ash.read!()
    |> Enum.map(&{Map.get(&1, label_field), &1.id})
  end
```

I can't seem to make the field name in the contains function dynamic. It works as expected if I replace ^label_field with an actual field name. I'm not sure if I should be turning to a custom action here, or if I'm just missing some syntax with the current approach. Any insight is appreciated.

**Comment by zachdaniel** [comment]
`label_field` should be `^ref(label_field)`

**Comment by netProphET** [comment]
Excellent! Thanks. Guess I should rtfm some more ;-)

**Pull request closed without merging**